proposal/custom-url-resolution.md

```diff
@@ -0,0 +1,230 @@
+# Custom url resolution: Draft 1
+
+_[(Issue)](https://github.com/sass/sass/issues/2535)_
+
+## Table of Contents
+
+- [Background](#background)
+- [Summary](#summary)
+  - [Example](#example)
+- [Function](#function)
+  - [Steps](#steps)
+  - [JavaScript API](#javascript-api)
+- [Presets](#presets)
+  - [The inline Preset](#the-inline-preset)
+    - [Inline Preset Usage](#inline-preset-usage)
+    - [Inline Preset Implementation](#inline-preset-implementation)
+  - [The relative Preset](#the-relative-preset)
+    - [Relative Preset Usage](#relative-preset-usage)
+    - [Relative Preset Implementation](#relative-preset-implementation)
+
+## Background
+
+> This section is non-normative.
+
+Many css features require the use of a sass-url reference to reference resources from outside the sass files, however these files also need to exist on the eventual output directory and server. To ensure the references are valid, the sass API should allow for the user to provide a way to remap and/or inline these resources.
+
+## Summary
+
+> This section is non-normative.
+
+This proposal defines a standardized way to remap the sass-url references to the final location on the server or inline reference.
+
+This is accomplished by running sass-url references through the url rewriting plugin if a url rewriting plugin has been defined.
```

Comment:
What is "a URL-rewriting plugin"? Where and how are they defined? Are there any built-in ones?

This is just the summary so it doesn't need to be meticulously precise, but this is extremely vague. You need to give a reader a sense of the specifics of your proposal so that they can evaluate it without needing to read the entire spec, not just say "here's the problem, this proposal will solve it". Think of this as a draft of the user-facing documentation that will explain how to use the new feature.